pkg/ingester/ingester.go

```diff
@@ -526,27 +529,13 @@ func (i *Ingester) starting(ctx context.Context) (err error) {
 	}
 
 	if i.ownedSeriesService != nil {
-		// We need to perform the initial computation of owned series after the TSDBs are opened but before the ingester becomes
-		// ACTIVE in the ring and starts to accept requests. However, because the ingester still uses the Lifecycler (rather
-		// than BasicLifecycler) there is no deterministic way to delay the ACTIVE state until we finish the calculations.
-		//
-		// Since we don't actually need to be ACTIVE in the ring to calculate owned series (just present, with tokens) we instead
-		// calculate owned series before we start the lifecycler at all. Once we move the ingester to the BasicLifecycler and
-		// have better control over the ring states, we could perform the calculation while in JOINING state, and move to
-		// ACTIVE once we finish.
+		// Start owned series service asynchronously. We don't wait for ownedSeriesService to enter Running state here.
```

Comment:
Since the OSS `starting` no longer has to wait for any external state to enter `Running` (it just skips the check if the ring is empty), should we also `AwaitRunning` for the OSS here? Since we haven't yet started the lifecycler, it shouldn't cause any state transition issue for the ingester (the ingester `starting` will take longer, but there should be no additional delay between the ring ACTIVE and the ingester `Running`).

Without explicitly waiting for the OSS to finish `starting`, and rejecting writes until it does, I think we run the risk of write failures during rollouts if the owned series calculation takes a long time: 
- `zone 1` ingesters are restarted, move to `ACTIVE`, and reject writes while the owned series are calculated
- `zone 2` ingesters are restarted because the rollout operator sees `zone 1` as healthy
- writes start to fail because two zones are unavailable/failing

I think waiting on the OSS `Running` doesn't have any adverse effect, and it would protect us against failing writes while the ingester is ACTIVE on startup. It would also simplify the code a bit, since we don't have to worry about situations where the ingester is `Running` but the OSS is not. What do you think?

Reply:
> I think waiting on the OSS `Running` doesn't have any adverse effect, and it would protect us against failing writes while the ingester is ACTIVE on startup. It would also simplify the code a bit, since we don't have to worry about situations where the ingester is `Running` but the OSS is not. What do you think?

This is now implemented in latest iteration of the PR.